hideshownetflixsubs.js

```diff
@@ -7,23 +7,11 @@
 // @match        https://www.netflix.com/*
 // @grant        none
 // ==/UserScript==
+'use strict';
 
-(function() {
-    'use strict';
-    function hideSubtitle(){
-        var hidecss = ".player-timedtext-text-container { display: none !important}";
-        if ((typeof document.subtitleVisible == "undefined") || document.subtitleVisible === true) {
-            document.styleSheets[0].insertRule(hidecss);
-            document.subtitleVisible = false;
-        } else
-        {
-            document.styleSheets[0].removeRule(hidecss);
-            document.subtitleVisible = true;
-        }
-    }
-    document.addEventListener('keydown', function(e) {
-        if (e.keyCode == 86 && !e.shiftKey && !e.ctrlKey && !e.altKey && !e.metaKey) {
-            hideSubtitle();
-        }
-    }, false);
-})();
+document.addEventListener('keydown', e => {
+  if(e.keyCode !== 86) return;
+
+  const subtitle = document.querySelector('.player-timedtext-text-container');
```

Comment:
Your approach has a bug, that it doesn't hide all subs. You are modifying, the DOM of the html. And you are using document.querySelector. This only returns first element that matches. You need to do document.querySelectorAll and loop over all the elements. In my approach I am changing the stylesheet of the document instead of adding style for each dom elements. Since these elements are dynamically added/removed (based on my observation) modifying the stylesheet approach is better.

Reply:
That makes sense . Thanks

Reply:
I started with dom approach myself. It wasn't working :P